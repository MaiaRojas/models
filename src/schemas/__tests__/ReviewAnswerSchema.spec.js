const mongoose = require('mongoose/browser');
const { ReviewQuestionSchema } = require('../')(mongoose);
const ReviewAnswerSchema = require('../ReviewAnswerSchema')(mongoose);


describe('ReviewAnswerSchema', () => {
  it('should fail validation when missing fields are provided', () => {
    const doc = new mongoose.Document({}, ReviewAnswerSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should successfully validate with proper values', (done) => {
    const question = new mongoose.Document({}, ReviewQuestionSchema);
    const doc = new mongoose.Document({
      question: question._id,
      value: 'Crack',
    }, ReviewAnswerSchema);

    return doc.validate((err) => {
      expect(err).toBe(null);
      done();
    });
  });
});
