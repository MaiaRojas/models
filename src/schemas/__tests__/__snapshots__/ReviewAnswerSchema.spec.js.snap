// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewAnswerSchema should fail validation when missing fields are provided 1`] = `
Object {
  "question": [ValidatorError: Path \`question\` is required.],
}
`;
